voucherify-service/src/__tests__/api-extension.service.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ConfigModule } from '@nestjs/config';
+import { Logger } from '@nestjs/common';
+import { RequestJsonLogger } from '../configs/requestJsonLogger';
+import { CommercetoolsConnectorService } from '../commercetools/commercetools-connector.service';
+import { ApiExtensionService } from '../commercetools/api-extension.service';
+import {
+  getCommerceToolsConnectorServiceMockForAPIExtensionServiceTest,
+  getConfigForAPIExtensionServiceTest,
+} from '../commercetools/__mocks__/commerce-tools-connector.service';
+import { MockFunctionMetadata, ModuleMocker } from 'jest-mock';
+
+const moduleMocker = new ModuleMocker(global);
+
+describe('ApiExtensionService', () => {
+  let service: ApiExtensionService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: ConfigModule,
+          useValue: getConfigForAPIExtensionServiceTest(),
+        },
+        ApiExtensionService,
+        {
+          provide: CommercetoolsConnectorService,
+          useValue:
+            getCommerceToolsConnectorServiceMockForAPIExtensionServiceTest(),
+        },
+        Logger,
+        RequestJsonLogger,
+      ],
+    })
+      .useMocker((token) => {
+        if (typeof token === 'function') {
+          const mockMetadata = moduleMocker.getMetadata(
+            token,
+          ) as MockFunctionMetadata<any, any>;
+          const Mock = moduleMocker.generateFromMetadata(mockMetadata);
+          return new Mock();
+        }
+      })
+      .compile();
+
+    service = module.get<ApiExtensionService>(ApiExtensionService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('should be defined', () => {
+    expect(service.list).toBeDefined();
+    expect(service.removeById).toBeDefined();
+    expect(service.removeByAttr).toBeDefined();
+    expect(service.add).toBeDefined();
+    expect(service.update).toBeDefined();
+  });
+
+  it('should be defined', async () => {
+    const result = await service.list();
+    expect(result).toBeDefined();
+  });
+
+  it('should be pass with no error', async () => {
```

Comment:
Pointless test:
- mock `ctConnector.extensions().get().execute()` used by `this.list()`
- test case when id not found (should return false)
- test case when id was found and returned (in addition, mock:  

```
this.client
        .extensions()
        .withId({ ID: extensionToRemove.id })
        .delete({ queryArgs: { version: extensionToRemove.version } })
        .execute()
```